src/main/java/pages/BasePage.java

```diff
@@ -35,11 +32,14 @@ public class BasePage {
     public By newsletterInputLocator = By.xpath(footersocialXpath + "//input[@id='appendedInputButton']");
     public By newsletterButtonLocator = By.xpath(footersocialXpath + "//button[@class='btn btn-orange']");
 
-    public BasePage() {
-        this.browserName = getBrowserName();
-        int baseWaitInSeconds = Integer.parseInt(PropertyManager.getProperty(PropertyManager.PropertyKeys.BASE_WAIT_IN_SECONDS));
-        this.wait = new WebDriverWait(getWebDriverInstance(), Duration.ofSeconds(baseWaitInSeconds));
-        this.baseUrl = PropertyManager.getProperty(PropertyManager.PropertyKeys.BASE_URL);
+    public BasePage(WebDriver driver) {
+        String someProperty = null; // get the property value
```

Comment:
I don't get this change:(

Reply:
@TOM-AN1  you can remove that change :)

Reply:
Please restore the previous constructor for BasePage!
```java
    public BasePage() {
        this.browserName = getBrowserName();
        int baseWaitInSeconds = Integer.parseInt(PropertyManager.getProperty(PropertyManager.PropertyKeys.BASE_WAIT_IN_SECONDS));
        this.wait = new WebDriverWait(getWebDriverInstance(), Duration.ofSeconds(baseWaitInSeconds));
        this.baseUrl = PropertyManager.getProperty(PropertyManager.PropertyKeys.BASE_URL);
    }
```
The change you made causes many errors! When calling methods from `BasePage`. e.g. `protected WebDriverWait wait = null` and therefore searching for elements does not work correctly (at all)

Reply:
Thank you. You helped me a lot. I thought I had undone this change. I rolled back the commit but forgot to check if I restored everything correctly.